core/modules/replica/Common.h

```diff
@@ -114,7 +114,7 @@ struct ReplicationRequestParams {
     unsigned int chunk;
     std::string  sourceWorker;
 
-    ReplicationRequestParams();
+    ReplicationRequestParams() = default;
```

Comment:
The default ctor used to initialize member variables, are you sure it's safe to not be doing that? Maybe you want to init them in the body of the class if you're going to `default` the ctor? (Check the other classes you changed too, I won't add noisy comments on all of them)

Reply:
This is safe because what I was doing earlier in the removed constructor was doing what the C++11 are supposed to do by the Standard:
- initializing strings with empty strings
- zero value initialization of the numeric fields
These are the members:
```
    int          priority;
    std::string  database;
    unsigned int chunk;
    std::string  sourceWorker;
```

Reply:
The default version of the c-tor would effectively do:
```
    int          priority = 0;
    std::string  databaser = std::string();
    unsigned int chunk = 0U;
    std::string  sourceWorker = std::string();
```

Reply:
oh, that's a change in C++11 I wasn't aware of. cool.

Reply:
Neither was I 😸 Andy-s once mentioned it to me when reviewing my code. Here is what the Standard says:
https://en.cppreference.com/w/cpp/language/zero_initialization

Reply:
This is actually a very interesting (an important) topic. I've just revisited the documentation on the C++11 (or later) and, I think, I'm convinced that for non-static members of classes, th default constructor will always perform zero-value initialization. See case 3 of following document:
https://en.cppreference.com/w/cpp/language/default_initialization
Which says:
```
Default initialization is performed in three situations:
...
3) when a base class or a non-static data member is not mentioned in a constructor initializer list and that constructor is called.
```
The second document explains what the default constructor does (how it initializes non-static members of the class):
https://en.cppreference.com/w/cpp/language/default_constructor
According to that document:
```
ClassName() = default;
```
is the same as:
```
ClassName() { }
```
And, for the missing initializations, zero value initialization would kick-in.

Obviously, I'm not a C++ lawyer 😸 So, I may be missing something here.

Reply:
For now I'm putting the explicit initializers (as suggested by @andy-slac ) back. Just in case.

Reply:
You are right about point 3 above, but it says that default initialization applies in that case, not zero initialization. Default initialization for basic types without constructor means no initialization. In the same document few lines below:

  nothing is done: the objects with automatic storage duration (and their subobjects) are initialized to indeterminate values.